### PR TITLE
[README] Add instruction for package installation for developers

### PR DESCRIPTION
**Pull request by XinweiHe**
As titled.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,6 +29,18 @@ You can install RelBench using `pip`:
...
 
+# Development
+
+If you wish to develop new features for RelBench:
+```
+python3 -m venv venv
+source venv/bin/activate
+pip install --upgrade pip
+pip install torch-scatter torch-sparse -f https://data.pyg.org/whl/torch-${TORCH}+${CUDA}.html
+pip install pyg-lib -f https://data.pyg.org/whl/torch-${TORCH}+${CUDA}.html
```
Comment (weihua916) [inline README.md:40]: No need.

File: setup.py
Comment (rusty1s) [inline setup.py:1]: No need to add a `setup.py` - there exists a `pyproject.toml` file already.

**Review by rusty1s — CHANGES_REQUESTED** [review]

**Comment by XinweiHe** [comment]
Without `setup.py` or `setup.cfg`, `pip install -e .` wouldn't work. Is there a more convenient way to install rel-bench in editable mode

**Comment by rusty1s** [comment]
It certainly does. What's the error you are seeing?

**Comment by XinweiHe** [comment]
> I think you may need to upgrade pip to latest version. old version does not work with pyproject.toml

Ah, I see. Let me close this PR then

**Pull request closed without merging**